segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -0,0 +1,210 @@
+     * @return                  CompletableFuture which when completed will have the segment registered on to the container
+     *                          metadata.
+     */
+    private static CompletableFuture<Void> recoverSegment(DebugStreamSegmentContainer container, SegmentProperties storageSegment) {
+        Preconditions.checkNotNull(container);
+        Preconditions.checkNotNull(storageSegment);
+        long segmentLength = storageSegment.getLength();
+        boolean isSealed = storageSegment.isSealed();
+        String segmentName = storageSegment.getName();
+
+        log.info("Registering: {}, {}, {}.", segmentName, segmentLength, isSealed);
+        return Futures.exceptionallyComposeExpecting(
+                container.getStreamSegmentInfo(storageSegment.getName(), TIMEOUT)
+                        .thenAccept(e -> {
+                            if (segmentLength != e.getLength() || isSealed != e.isSealed()) {
+                                container.metadataStore.deleteSegment(segmentName, TIMEOUT)
+                                        .thenAccept(x -> container.registerSegment(segmentName, segmentLength, isSealed));
+                            }
+                        }), ex -> Exceptions.unwrap(ex) instanceof StreamSegmentNotExistsException,
+                () -> container.registerSegment(segmentName, segmentLength, isSealed));
+    }
+
+    /**
+     * Deletes container metadata segment and its Attribute segment from the {@link Storage} for the given container Id.
+     * @param storage       A {@link Storage} instance to delete the segments from.
+     * @param containerId   Id of the container for which the segments has to be deleted.
+     */
+    public static CompletableFuture<Void> deleteMetadataAndAttributeSegments(Storage storage, int containerId) {
+        Preconditions.checkNotNull(storage);
+        String metadataSegmentName = NameUtils.getMetadataSegmentName(containerId);
+        String attributeSegmentName = NameUtils.getAttributeSegmentName(metadataSegmentName);
+        return deleteSegmentFromStorage(storage, metadataSegmentName)
+                .thenAccept(x -> deleteSegmentFromStorage(storage, attributeSegmentName));
+    }
+
+    /**
+     * Deletes the segment with given name from the given {@link Storage} instance. If the segment doesn't exist, it does
+     * nothing and returns.
+     * @param storage       A {@link Storage} instance to delete the segments from.
+     * @param segmentName   Name of the segment to be deleted.
+     * @return              CompletableFuture which when completed will have the segment deleted. In case segment didn't
+     *                      exist, a completed future will be returned.
+     */
+    private static CompletableFuture<Void> deleteSegmentFromStorage(Storage storage, String segmentName) {
+        log.info("Deleting Segment '{}'", segmentName);
+        return Futures.exceptionallyComposeExpecting(
```

Comment:
use `exceptionallyExpecting` and return `null` instead of `CompletableFuture.completedFuture(null)`

Reply:
Ok.